### DISCUSSION TITLE
Use of plugins with Windows

### DISCUSSION THREAD

**Discussion by Messtechniker** [rejetto/hfs#483] [General]
Created the following to run antidos:
HFS Home directory
--> Plugins
         -->antidos
                   --> plugin.js
Trying to run plugin.js yields:
Line 1
Char 11
Error ';' expected 
800A03EC
Compilation error in MS JScript.
What should I do?

**Comment by rejetto** [comment]
you should install and run plugins using the interface in Admin-panel > plugins

**Comment by rejetto** [comment]
plugins must not be launched like any program, they don't run on their own, they must be started by hfs

**Comment by Messtechniker** [comment]
Thanks for your quick response.
Searched the menu in expert mode up and down for the admin panel and could not find it.
Sorry.

**Comment by rejetto** [comment]
expert mode?
this is HFS 3 here, you are probably using HFS 2

**Comment by rejetto** [comment]
HFS 2 has no "plugins", and is not compatible with the stuff of hfs 3.
Although "templates" in v2 were so powerful that they could be considered like plugins, but still not compatible with v3.

**Comment by Messtechniker** [comment]
Yes. I'm on 2.3m. Which is announced on the home page as the latest version. Hmmm...
Since 2.3m is running fine, I'll leave it as it is for the moment. 
Because it is being hammered quite regularly, I will be running hfs only on demand.

Thankyou for your effort.

By the way:
On the home page there is no version 3 to download. Only a preview version 2.4.0 RC6

**Comment by rejetto** [comment]
2.3 is latest official, yes, but this is a different website, where i'm working on the next release.
You can learn more about it reading here
https://github.com/rejetto/hfs/?tab=readme-ov-file#hfs-http-file-server-version-3

HFS 1-2 were designed to be "occasional" servers, while 3 is meant to be always-on.

**Comment by Messtechniker** [comment]
Just digging a bit into Version 3. Very nice. 
In my opinion, easier to set up compared to Version 2.
Even got antidos working very quickly too.
:-)

**Comment by rejetto** [comment]
it's nice to hear!
cleary v3 is very different, i wasn't even sure if to keep the name

**Comment by Messtechniker** [comment]
Very different indeed. But easy. Was able to customise it in no time. Header and login.
And I am not the best programming guy on the planet. Very much prefer doing hardware 
repairs on electronics of all kinds.
Since I am running HFS 24/7 on an old, but energy-wise not so thirsty Windows 7 notebook, 
I will have to stick with 2.3m, since Version 3 requires Windows 8.1+. 
Does not really bother me, because 2.3m ist running fine on Windows 7.

**Comment by rejetto** [comment]
thanks for the feedback.
you can reconsider v3 if you'll change laptop or operating system, as it runs on linux too.
